test/eth.subscribe.ganache.js

```diff
@@ -130,6 +130,7 @@ describe('subscription connect/reconnect', function () {
             .subscribe('newBlockHeaders')
             .on("error", function (err) {
                 assert(err.message.includes('No provider set'));
+                this.removeAllListeners();
```

Comment:
Why exactly is here in this test always ``this.removeAllListeners()`` required?

Reply:
I'm seeing the listeners continue to fire in the background sometimes. Esp. noticeable when  `pify(server.close)()` is called on a closed server - an unhandled rejection gets thrown. 

Have done this to all of them as "best practice" to clean-up on finish.

Reply:
You could also use ``once`` instead of ``on`` which will do the same :)

Reply:
Ok I'll go through and see which one's can be once - some of them probably can't because they are multi-call or multi-listener.